src/client.js

```diff
@@ -148,15 +145,31 @@ assign(Client.prototype, {
 
   Pool: Pool2,
 
-  initializePool: function(config) {
-    if (this.pool) this.destroy()
-    this.pool = new this.Pool(assign(this.poolDefaults(config.pool || {}), config.pool))
-    this.pool.on('error', function(err) {
-      helpers.error('Pool2 - ' + err)
-    })
-    this.pool.on('warn', function(msg) {
-      helpers.warn('Pool2 - ' + msg)
+  initializePool: function(config, callback) {
+    var client = this
+    var promise = Promise.try(function() {
+      return Promise.try(function() {
+        if (client.pool) {
+          return client.destroy()
+        }
+      })
+      .then(function() {
+        client.pool = new client.Pool(assign(client.poolDefaults(config.pool || {}), config.pool))
+        client.pool.setMaxListeners(0) // Suppress warnings about too many event listeners
```

Comment:
Is it possible to set it to a high number that will still be triggered when something goes wrong, without it being triggered in the course of normal proceedings? I assume this event is only used internally, so we might be able to calculate the max number of listeners from max connections?